server/permissionValidation/course.go

```diff
@@ -8,14 +8,14 @@ import (
 	"github.com/google/uuid"
 )
 
-func CheckCoursePermission(c *gin.Context, courseID uuid.UUID, allowedUsers []string, adminRight ...string) (bool, error) {
+func CheckCoursePermission(c *gin.Context, courseID uuid.UUID, allowedUsers []string) (bool, error) {
```

Comment:
I would make this as a Variadic and not as a string slice
```suggestion
func CheckCoursePermission(c *gin.Context, courseID uuid.UUID, allowedUsers ...string) (bool, error) {
```